app/src/main/java/com/hedvig/app/feature/referrals/ui/tab/ReferralsAdapter.kt

```diff
@@ -218,6 +231,26 @@ class ReferralsAdapter : RecyclerView.Adapter<ReferralsAdapter.ViewHolder>() {
                     }
             }
         }
+
+        class ErrorViewHolder(parent: ViewGroup) : ViewHolder(
+            LayoutInflater
+                .from(parent.context)
+                .inflate(R.layout.referrals_error, parent, false)
+        ) {
+            private val retry = itemView.retry
+            override fun bind(data: ReferralsModel, reload: () -> Unit) {
+                retry.setHapticClickListener { reload() }
+            }
+        }
     }
-}
 
+    companion object {
+        val LOADING_STATE = listOf(
```

Comment:
private

Reply:
It's used by `ReferralsFragment`. I guess I could make it a method of the `ReferralsAdapter`, though!